.github/pull_request_template.md

```diff
@@ -16,24 +16,45 @@ Bug Fixes
 
 Checklist
 ---------
-- [ ] Leave this box unchecked if features are not yet available in production
+<!-- 
+Check each item below to ensure high-quality CLI development practices are followed. PR approval will not be granted until the checklist is fully reviewed.
+For detailed instructions, please refer to this Confluence page: https://confluentinc.atlassian.net/wiki/spaces/AEGI/pages/3949592874/
+-->
+- [ ] I have successfully built and used a custom CLI binary, with no linting issues identified from this PR.
+- [ ] I have clearly specified in the `Release Notes` section above whether this PR applies to Confluent Cloud, Confluent Platform, or both. 
+- [ ] I have verified this PR with Confluent Cloud resources in a production environment.
+- [ ] I have verified this PR with Confluent Platform resources in an on-premises environment.
```

Comment:
Nitpick: we could indicate that we only need to check ones that apply; cloud only changes won't need to verify manually verify on-prem.

Reply:
Maybe 
`I have verified this PR with Confluent Cloud resources in a production environment, if applicable.`
`I have verified this PR with Confluent Platform resources in an on-premises environment, if applicable.`

Reply:
I like it